src/engine/MaxConstraint.cpp

```diff
@@ -481,23 +602,43 @@ void MaxConstraint::updateVariableIndex( unsigned oldIndex, unsigned newIndex )
         _f = newIndex;
 
     if ( _elements.exists( oldIndex ) )
-	{
-	    _elements.erase( oldIndex );
-	    _elements.insert( newIndex );
-	}
+    {
+        _elements.erase( oldIndex );
+        _elements.insert( newIndex );
+    }
 }
 
 bool MaxConstraint::constraintObsolete() const
 {
     return _obsolete;
 }
 
-void MaxConstraint::eliminateVariable( unsigned var, double /*value*/ )
-{
+void MaxConstraint::eliminateVariable( unsigned var, double value )
+{ // todo updated - Guy A
```

Comment:
Here, also update _maxLowerBound if we need to. This will make it so that if you have maxEliminatedValue = 5 and some other variable x's upper bound gets updated to < 5, than x will be removed (in notify upper bound).